rust/chains/hyperlane-cardano/src/rpc/mod.rs

```diff
@@ -51,6 +55,100 @@ impl CardanoRpc {
         merkle_tree(&self.0).await
     }
 
+    pub async fn get_ism_parameters(
+        &self,
+    ) -> Result<InboxIsmParameters200Response, Error<InboxIsmParametersError>> {
+        inbox_ism_parameters(&self.0).await
+    }
+
+    pub async fn is_inbox_message_delivered(
+        &self,
+        message_id: H256,
+    ) -> Result<IsInboxMessageDelivered200Response, Error<IsInboxMessageDeliveredError>> {
+        is_inbox_message_delivered(&self.0, message_id.encode_hex::<String>().as_str()).await
+    }
+
+    // NOTE: We must mock so much we don't even need a tight/deterministic
+    // implementation in the RPC...
+    pub async fn estimate_inbox_message_fee(
+        &self,
+        message: &HyperlaneMessage,
+        metadata: &[u8],
+    ) -> Result<EstimateInboundMessageFee200Response, Error<EstimateInboundMessageFeeError>> {
+        let origin_mailbox = format!("0x{}", hex::encode(&metadata[0..32]));
+        let checkpoint_root = hex::encode(&metadata[32..64]);
+        let checkpoint_index = u32::from_be_bytes(metadata[64..68].try_into().unwrap());
+        let signatures = metadata[68..metadata.len() - 1]
+            .chunks(64)
+            .map(|s| hex::encode(s))
+            .collect();
+        estimate_inbound_message_fee(
+            &self.0,
+            EstimateInboundMessageFeeRequest {
+                relayer_cardano_address:
```

Comment:
I'd suggest moving the relayer private key and address configs to RPC ENV variables.

In the current architecture, RPC service is run by relayer operator and is trustful.

We don't have capacity for full Ed25519 signing support on relayer, but can later address that in https://github.com/tvl-labs/hyperlane-cardano/issues/51